### ISSUE TITLE
Python syntax highlighting is worse after the latest update

### ISSUE THREAD

**Issue by dgudim** [jdinhify/vscode-theme-gruvbox#82]
Before (variables, function calls and definitions are kinda separated)
![image](https://github.com/jdinhify/vscode-theme-gruvbox/assets/34401005/1152f4ad-6c83-4233-97dc-074bcd089157)

After (everything is blue and yellow now, even less separation)
![image](https://github.com/jdinhify/vscode-theme-gruvbox/assets/34401005/324c58bd-ecae-4cbb-a05e-fcb3fba10958)

**Comment by song-dog** [comment]
@dgudim I just spent the last 3 hours banging my head against the wall trying to figure out why all of a sudden my syntax highlighting had suddenly changed. I believe I finally traced the culprit back to this extension, and specifially the update from yesterday re: semantic highlighting. Once i switched that setting from "configuredByTheme" to "false", everything went back to normal.
<img width="375" alt="semanticHighlighting-0" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/132631979/04334ad6-6ee8-451b-b5a9-387ce13094d9">
Alternatively you can override the semantic highlighting rules yourself if you prefer.
<img width="419" alt="semanticHighlighting-1" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/132631979/0be4539c-0068-47a0-b868-004f7540d853">

**Comment by dgudim** [comment]
@song-dog oh my god, thanks. The theme is probably missing colors for semantic token scopes

**Comment by jdinhify** [comment]
@song-dog apologies for the inconvenience

there's another PR to enhance these semantic token, but I think it won't address your issue

so could you please try either these settings and let me know which would be more helpful to you 🙏 

1.

```
"editor.semanticTokenColorCustomizations": {
  "rules": {
    "builtinConstant": "#d3869b",
    "property": "#8ec07c",
    "parameter": "#83a598",
    "variable": "#ebdbb2",
    "magicFunction": "#fe8019",
    "function": "#fe8019",
    "method": "#fe8019"
  }
},
```
<img width="2032" alt="Screen Shot 2023-10-26 at 6 37 04 pm" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/14046273/5bec0d2e-8444-452b-b6e2-999aba362761">

2.
```
"editor.semanticTokenColorCustomizations": {
  "rules": {
    "builtinConstant": "#d3869b",
    "property": "#ebdbb2",
    "parameter": "#83a598",
    "variable": "#8ec07c",
    "magicFunction": "#fe8019",
    "function": "#fe8019",
    "method": "#fe8019"
  }
},
```
<img width="2032" alt="Screen Shot 2023-10-26 at 6 37 32 pm" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/14046273/eaf2aed6-ab97-4521-8e16-8e00bed17e65">

**Comment by dgudim** [comment]
@jdinhify The first one looks better IMO

**Comment by jdinhify** [comment]
this has been merged & deploy #83

**Comment by dgudim** [comment]
Can we differentiate between 'magic' and regular functions? (__file__ and main) for example?

**Comment by jdinhify** [comment]
> Can we differentiate between 'magic' and regular functions? (**file** and main) for example?

I work mainly with JS/TS & don't see these magic functions often (or ever) so not sure about the importance of the distinction. In python, looks like the `__` wrappers are already used to differentiate?

What would you think is a good colour if have to choose? Colours to pick below - I tried them and thought that the current setting (same colour) is the most ok:

```
fb4934
fabd2f
83a598
d3869b
8ec07c
fe8019
```

**Comment by dgudim** [comment]
Ok, maybe leave it as default then, I'll change it locally, probably to darker orange or dark magenta

**Comment by dgudim** [comment]
I have 1 more question though, even without semantic stuff 
![image](https://github.com/jdinhify/vscode-theme-gruvbox/assets/34401005/f980a049-380e-4f93-bf02-11300b6e838b)

Some of the stuff is orange now, format string was red, number was white and False was purple

**Comment by jdinhify** [comment]
> I have 1 more question though, even without semantic highlighting ![image](https://user-images.githubusercontent.com/34401005/278610205-f980a049-380e-4f93-bf02-11300b6e838b.png)
> 
> Some of the stuff is orange now, format string was red, number was purple and False was purple

There a new version published, it should be more aligned with pre-semantic highlighting, please have another try 🙏

**Comment by jdinhify** [comment]
> And this is also orange now, was red. Although both variants look bad ![image](https://user-images.githubusercontent.com/34401005/278611402-7b5a8264-e867-4438-b198-5259ef683382.png)

there was never a background set for matchHighlight in peekView
a new version has been published, using gray as background for more neutral highlighting

**Comment by dgudim** [comment]
@jdinhify 
It's almost perfect ❤️
I have some suggestions though
1. Make modules a different shade of yellow, maybe a little darker
2. Make builtin variables purple or orange instead of white
3. Make magic functions/variables different color from regular functions/variables

![image](https://github.com/jdinhify/vscode-theme-gruvbox/assets/34401005/a58871e4-12d2-4b6f-a511-225b68c0ec52)

And also, where can I donate?

**Comment by alecdwm** [comment]
I wasn't a huge fan of the new colours in the `v1.16.0` version of Gruvbox for `Typescript JSX`, so I added two overrides:

```json5
"editor.semanticTokenColorCustomizations": {
  "[Gruvbox Dark Medium]": {
    // use the same variable and function colors for semantic syntax highlighting `on` and `off`
    "rules": { "variable": "#83a598", "function": "#fabd2f" }
  }
},
```

<div align="center">

<img width="680" alt="Screenshot 2023-10-31 at 02 38 05" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/2741569/536f0cc3-7aeb-4903-a6fb-05d195e1bf1c">

*semantic highlighting disabled*

---

<img width="680" alt="Screenshot 2023-10-31 at 02 38 25" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/2741569/e92fbbef-edc3-4277-8d03-623540fee44e">

*semantic highlighting enabled (default colors)*

---

<img width="680" alt="Screenshot 2023-10-31 at 02 38 34" src="https://github.com/jdinhify/vscode-theme-gruvbox/assets/2741569/0b31a9ba-ba69-4711-9aac-9b1571b17dfb">

*semantic highlighting enabled (with color overrides)*

</div>

**Comment by dgudim** [comment]
That's not python though

**Comment by alecdwm** [comment]
Yeah! My comment is only relevant to the new syntax highlighting, not specifically how it affects python.  
That being said, the changes should apply to any language.

**Comment by jdinhify** [comment]
thanks @alecdwm 

there's a discussion for TS in #85 , let's discuss over there

I found out in the past few days that semantic tokens colouring can be applied to specific languages, and I'm not a python user so I'd respect the input of python users, so will try to make changes to TS (probably the default since I'm using TS too) while not affecting python as requested here

**Comment by jdinhify** [comment]
> @jdinhify It's almost perfect ❤️ I have some suggestions though
> 
> 1. Make modules a different shade of yellow, maybe a little darker
> 2. Make magic functions/variables different color from regular functions/variables (or maybe italic?)
> 
> ![image](https://user-images.githubusercontent.com/34401005/278805928-a58871e4-12d2-4b6f-a511-225b68c0ec52.png)
> 
> And also, where can I donate?

thanks @dgudim ,

1. I'm not familiar with python, what are these "modules"? Could you provide an example snippets
2. Would you think a darker version of functions would be ok? Variables might be trickier since vscode doesn't have a `magicVariable`, just `builtin` so it will affect all builtin variables, but I guess `builtin` are `magic`. If we're using a darker version for magic functions, I think we'll probably use a darker version for builtin variables too

**Comment by dgudim** [comment]
@jdinhify 
1. 
![image](https://github.com/jdinhify/vscode-theme-gruvbox/assets/34401005/bb0b608b-8416-4170-8dfa-58629e47fff9)
Here are the modules
2. I was thinking an italic version with the same color for magic functions and italic purple for magic variables. As for the scope, there is `support.variable.magic.python` textmate scope, but maybe coloring all builtins would also be fine
